spark/src/test/scala/org/apache/spark/sql/delta/DeltaSourceSuite.scala

```diff
@@ -1774,10 +1774,13 @@ class DeltaSourceSuite extends DeltaSourceSuiteBase
           q.processAllAvailable()
           checkAnswer(
             sql("select * from startingVersionTest"),
-            ((10 until 20).map(x => (x.toLong, x.toLong, None.toString)) ++
+            ((10 until 20).map(x => (x.toLong, x.toLong, "null")) ++
               (20 until 30).map(x => (x.toLong, x.toLong, x.toString)))
               .toDF("id", "id2", "id3")
-              .selectExpr("id", "id2", "cast(id3 as long) as id3")
+              .selectExpr(
+                "id",
+                "id2",
+                "CASE WHEN id3 = 'null' THEN NULL ELSE cast(id3 as long) END as id3")
```

Comment:
Is the behavior of `cast` changed in Spark master? Why do we need this change?

Reply:
Yes it did change. Spark Master enables ANSI SQL mode by default